Make it possible to run tests without an internet connection

When I turn off my internet connection and run the test suite, I can see two failures. This should be fixed, so that the test suite is not dependent on the internet connection.

```
Running Suite: gobrake
======================
Random Seed: 1597204724
Will run 39 of 39 specs

••••••••••••
------------------------------
• Failure [0.003 seconds]
httpmetric
/Users/kyrylo/code/airbrake/gobrake/metric_test.go:129
  measures timing until first response byte [It]
  /Users/kyrylo/code/airbrake/gobrake/metric_test.go:130

  Unexpected error:
      <*url.Error | 0xc0002953e0>: {
          Op: "Get",
          URL: "https://www.google.com/",
          Err: {
              Op: "dial",
              Net: "tcp",
              Source: nil,
              Addr: nil,
              Err: {
                  Err: "no such host",
                  Name: "www.google.com",
                  Server: "",
                  IsTimeout: false,
                  IsTemporary: false,
                  IsNotFound: true,
              },
          },
      }
      Get "https://www.google.com/": dial tcp: lookup www.google.com: no such host
  occurred

  /Users/kyrylo/code/airbrake/gobrake/metric_test.go:138
------------------------------
••••••••••••••••••
------------------------------
• Failure [0.002 seconds]
Notifier
/Users/kyrylo/code/airbrake/gobrake/notifier_test.go:31
  logs errors on invalid project id or key [It]
  /Users/kyrylo/code/airbrake/gobrake/notifier_test.go:344

  Expected
      <string>: sendNotice failed reporting notice="Notice<*errors.errorString: oops>": Post "https://api.airbrake.io/api/v3/projects/1/notices": dial tcp: lookup api.airbrake.io: no such host

  to contain substring
      <string>: invalid project id or key

  /Users/kyrylo/code/airbrake/gobrake/notifier_test.go:363
------------------------------
•••••••

Summarizing 2 Failures:

[Fail] httpmetric [It] measures timing until first response byte
/Users/kyrylo/code/airbrake/gobrake/metric_test.go:138

[Fail] Notifier [It] logs errors on invalid project id or key
/Users/kyrylo/code/airbrake/gobrake/notifier_test.go:363

Ran 39 of 39 Specs in 0.360 seconds
FAIL! -- 37 Passed | 2 Failed | 0 Pending | 0 Skipped
--- FAIL: TestGobrake (0.36s)
FAIL
exit status 1
FAIL    github.com/airbrake/gobrake/v4  0.482s
```

--- a/metric_test.go
+++ b/metric_test.go
@@ -3,6 +3,7 @@ package gobrake
 import (
 	"context"
 	"net/http"
+	"net/http/httptest"
 	"sync"
 	"time"
 
@@ -127,11 +128,22 @@ var _ = Describe("metric with fake clock", func() {
 })
 
 var _ = Describe("httpmetric", func() {
+	var server *httptest.Server
+
+	BeforeEach(func() {
+		handler := func(w http.ResponseWriter, req *http.Request) {
+			w.WriteHeader(http.StatusNotFound)
+			_, err := w.Write([]byte(""))
+			Expect(err).To(BeNil())
+		}
+		server = httptest.NewServer(http.HandlerFunc(handler))
+	})
+
 	It("measures timing until first response byte", func() {
 		c := context.Background()
 		c, metric := NewRouteMetric(c, "GET", "/api/v1/projects/:projectId")
 
-		req, _ := http.NewRequest("GET", "https://www.google.com/", nil)
+		req, _ := http.NewRequest("GET", server.URL, nil)
 		req = req.WithContext(c)
 
 		_, err := http.DefaultClient.Do(req)

--- a/notifier_test.go
+++ b/notifier_test.go
@@ -362,11 +362,21 @@ var _ = Describe("Notifier", func() {
 		l := log.New(buf, "", 0)
 		gobrake.SetLogger(l)
 
+		// Return Unauthorized for the error API.
+		handler := func(w http.ResponseWriter, req *http.Request) {
+			w.WriteHeader(http.StatusUnauthorized)
+			_, err := w.Write([]byte(""))
+			Expect(err).To(BeNil())
+		}
+		errorServer := httptest.NewServer(http.HandlerFunc(handler))
+
 		configServer := newConfigServer()
+
 		n := gobrake.NewNotifierWithOptions(
 			&gobrake.NotifierOptions{
 				ProjectId:        1,
 				ProjectKey:       "broken-key",
+				Host:             errorServer.URL,
 				RemoteConfigHost: configServer.URL,
 			},
 		)

